onnxruntime/test/contrib_ops/col2im_test.cc

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <stdexcept>
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+#include "core/util/math.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace test {
+using namespace onnxruntime::test;
+
+namespace {
+template <typename T>
+std::vector<T> _transpose_serialized_vector(std::vector<T> &input, size_t N, size_t C, size_t H, size_t W) {
+    size_t input_size = input.size();
+    if (input_size == 0) {
+        throw std::runtime_error("Invalid input");
+    }
+    std::vector<T> trans_vec(input);
+
+    for (size_t n = 0; n < N; ++n)
+      for (size_t c = 0; c < C; ++c)
+        for (size_t h = 0; h < H; ++h)
+          for (size_t w = 0; w < W; ++w)
+              trans_vec[n * (C * H * W) + c * (H * W) + (h + H * w)] = \
+                input[n * (C * H * W) + c * (H * W) + (w + W * h)];
+
+    return trans_vec;
+}
+
+struct float_iota {
+    explicit float_iota(float inc, float init_value = 0.0) : _value(init_value), _inc(inc) {}
+
+    operator float() const { return _value; }
+    float_iota& operator++() { _value += _inc; return *this; }
+    float _value;
+    float _inc;
+};
+
+}  // namespace
+
+TEST(Col2ImContribOpTest, simple4dNCHW) {
+  OpTester test("Col2Im", 1, kMSDomain);
+
+  test.AddAttribute("strides", std::vector<int64_t>{1, 1});
+  test.AddAttribute("dilations", std::vector<int64_t>{1, 1});
+  test.AddAttribute("pads", std::vector<int64_t>{0, 0, 0, 0});
+
+  std::vector<float> input(25);
+  std::vector<float> output(25);
+  std::iota(output.begin(), output.end(), float_iota(1., 1.));
```

Comment:
std::iota should work directly with float too

```suggestion
  std::iota(output.begin(), output.end(), 1.0f);
```

Reply:
done